### PR TITLE
Enable client tls with given certificates

### PR DESCRIPTION
**Pull request by akihikokuroda**
Enable client tls with given certificates

Setting the secret into HELM_SECRET environment variable enables tls options for the helm commands.

The secret should have:

> cert.pem: certificate
> key.pem: certificate key
> ca.pam: CA certificate.

Sample secret creation command:

`create secret generic <secret name> --from-file=cert.pem=.helm/cert.pem --from-file=ca.pem=.helm/ca.pem --from-file=key.pem=.helm/key.pem`

### CODE REVIEW

**Comment by akihikokuroda** [comment]
The tls options are not applied to the helm init.  It can not be put into the variable.  I found some problems in the verify stage.  I fixed them.

**Comment by akihikokuroda** [comment]
I made the changes and finally made it work.

**Comment by davidcurrie** [comment]
The files that I see generated in .helm are called ca.pem, cert.pem and key.pem. Where did the tls.crt, tls.key and ca.crt names come from?

**Comment by a-roberts** [comment]
Missed that, and we shouldn't really be saying users are going to create the secret and rename their files before placing them in...

**Comment by akihikokuroda** [comment]
I updated the key names in the secret.

File: vars/microserviceBuilderPipeline.groovy [hunk 1]
```diff
@@ -301,17 +316,22 @@ def getDeployBranch () {
...
 
-def initalizeHelm (String tillerNamespace) {
-  container ('helm') {        
-    sh "helm init --skip-refresh"    
+def initalizeHelm (String tillerNamespace, String helmSecret) {
+  container ('helm') {
+    def options = ""
+    if (helmSecret) {
+       echo "initializing with --client only"
+       options += "--client-only "
```
Comment (davidcurrie) [inline vars/microserviceBuilderPipeline.groovy]: We should now always be` --client-only`. Jenkins should never be creating a Tiller.

**Comment by akihikokuroda** [comment]
I removed tillerNamespace and changed to always use --client-only.

**Pull request merged**